Source/CBLRouter+Changes.m

```diff
@@ -127,6 +138,15 @@ - (CBLStatus) doChanges: (CBLDatabase*)db {
                 heartbeat = kMinHeartbeat;
             NSString* heartbeatResponse = (_changesMode == kEventSourceFeed) ? @":\n\n" : @"\r\n";
             [self startHeartbeat: heartbeatResponse interval: heartbeat];
+        } else {
+            // Apply default timeout when heartbeat is not specified:
+            if (_changesTimeout == 0)
+                _changesTimeout = kDefaultChangesTimeout;
+        }
+        
+        if (_changesTimeout > 0) {
+            _changesTimeoutSince = since;
```

Comment:
I think it would be better to always store the 'since' value in an ivar instead of sometimes in a local. Otherwise if we ever forget to set `_changesTimeoutSince` before calling -startTimeout, it causes a bug.

So basically remove the local var `since` and replace it with the ivar `_changesTimeoutSince` ... actually rename that `_changesSince` because it's now used even without a timeout.